plugins/available/goenv.plugin.bash

```diff
@@ -0,0 +1,11 @@
+cite about-plugin
+about-plugin 'load goenv, if you are using it'
+
+# Don't modify the environment if we can't find the tool
+command -v goenv &>/dev/null ||
+  [[ -f "$HOME/.goenv/bin/goenv" ]] ||
+  return
+
+export GOENV_ROOT="$HOME/.goenv"
+pathmunge "$GOENV_ROOT/bin"
+eval "$(goenv init - bash)"
```

Comment:
If the check for the `goenv` command succeeds, its already on the path, and its highly likely that `GOENV_ROOT` is already defined as well.

Feels like we might only want to init if :
* `goenv` is not already on the path
* `GOENV_ROOT` is not already defined.
* `$HOME/.goenv/bin/goenv` exists

Anything else feels like it could be too complicated to handle the various present/not-present combinations correctly.

re `command -v goenv`:
Bashit provides [command_exists](https://github.com/Bash-it/bash-it/blob/7c09252ac86f430b9a6eaeebfc19b20a9f6fe3ce/plugins/available/base.plugin.bash#L163) to make these checks more convenient.

Reply:
@davidpfarrell  Thanks for the thoughtful review! I'm going to respond to points as I see them, and hope that helps makes sense of what I'm thinking. In the end, I think I have addressed all your points.

> If the check for the goenv command succeeds, its already on the path, and its highly likely that GOENV_ROOT is already defined as well.

The reason for checking if it's in the path already is because package managers (e.g. Homebrew) install `goenv` to a directory already in `PATH`. Unfortunately you are incorrect, `GOENV_ROOT` is still on the user to manage, since technically it's not a required value as goenv will default to `~/.goenv`.

> Feels like we might only want to init if

Unfortunately it's on the user to manage. I haven't found an example of any packages that configure the user's profile.

> `GOENV_ROOT` is not already defined.

Great catch. I definitely prefer to use optional assignment. my bad.

> `$HOME/.goenv/bin/goenv` exists

This presumption would mean it wouldn't work with Homebrew.

> Anything else feels like it could be too complicated to handle the various present/not-present combinations correctly.

After considering all this, I came up with a test for initialization. It's specific to this tool, but it works, because when `init` is run, `goenv` gets overloaded as a function; `type goenv 2>/dev/null | grep -q 'is a function'`.

Please have another look and tell me what you think. Thanks for the review!!

Reply:
Yes, the `helpers` function `_command_exists` should be used, since that can't be disabled.